test/ci-cd/scripts/comfyui/run_in_cluster.sh

```diff
@@ -0,0 +1,109 @@
+#   cluster_name, cluster_region, cluster_project_id, comfyui_kubernetes_namespace, comfyui_endpoints_hostname,
+#
+# ==============================================================================
+
+# --- Script Configuration ---
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source /workspace/build.env
+if [ "${DEBUG,,}" == "true" ]; then
+  set -o xtrace
+fi
+
+# Check for all required environment variables
+if [ -z "$cluster_name" ] || [ -z "$cluster_region" ] || [ -z "$cluster_project_id" ] || \
+   [ -z "$comfyui_kubernetes_namespace" ] || [ -z "$comfyui_endpoints_hostname" ] || [ -z "$WORKFLOWS_DIR" ] || \
+   [ -z "$TEST_DIR" ]; then
+    echo "Error: One or more required environment variables are not set." >&2
+    echo "Required: cluster_name, cluster_region, cluster_project_id, comfyui_kubernetes_namespace, HEADLESS_COMFYUI_SERVICE, WORKFLOWS_DIR, TEST_DIR" >&2
+    exit 1
+fi
+
+# --- Variables ---
+# Use a unique name for the pod for concurrent runs.
+POD_NAME="comfyui-client"
+printenv
+# --- verify each file has a test file ---
+check_files() {
+  for file in "$WORKFLOWS_DIR"/*; do
+    if [[ ! -f "$TEST_DIR/workflows/$(basename "$file")" ]]; then
+      echo "Error: $(basename "$file") not found in $TEST_DIR." >&2
+      exit 1
+    fi
+  done
+}
+# --- Cleanup Function ---
+cleanup() {
+  echo "--- Cleaning up pod: $POD_NAME ---"
+  kubectl delete pod "$POD_NAME" --namespace="$comfyui_kubernetes_namespace" --ignore-not-found=true || true
+}
+trap cleanup EXIT
+
+
+# 1. Get GKE cluster credentials
+echo "--- Getting GKE credentials for project '$cluster_project_id' ---"
+gcloud container clusters get-credentials "$cluster_name" --region "$cluster_region" --project "$cluster_project_id" --dns-endpoint
+
+# 2. Run a temporary client pod
+echo "--- Creating pod: $POD_NAME in namespace '$comfyui_kubernetes_namespace' ---"
+kubectl run "$POD_NAME" \
+  --image=alpine:latest \
+  --restart=Never \
+  --namespace="$comfyui_kubernetes_namespace" \
+  --command -- sh -c "apk add --no-cache curl jq && echo 'Pod is ready. Waiting...' && sleep 3600"
+
+# 3. Wait for the pod to be ready
+echo "--- Waiting for pod to be ready ---"
+kubectl wait --for=condition=Ready pod/"$POD_NAME" --namespace="$comfyui_kubernetes_namespace" --timeout=60s
+
+# 4. Copy files to the pod
+echo "--- Copying files to pod ---"
+kubectl cp "$TEST_DIR/comfyui_prompt_test.sh" "${comfyui_kubernetes_namespace}/${POD_NAME}:/tmp/comfyui_prompt_test.sh"
+kubectl cp "$TEST_DIR/workflows" "${comfyui_kubernetes_namespace}/${POD_NAME}:/tmp/workflows"
+
+# 5. Make the runner script executable inside the pod
+echo "--- Setting script permissions in pod ---"
+kubectl exec --namespace="$comfyui_kubernetes_namespace" "$POD_NAME" -- chmod +x /tmp/comfyui_prompt_test.sh 
+
+# 6. Get ComfyUI IP address
+echo "--- Getting Comfy IP ---"
+SERVICE_IP_AND_PORT=$(kubectl get service -n comfyui comfyui-nvidia-l4 -o=json | jq -r '"\(.spec.clusterIP):\(.spec.ports[0].port)"')
```

Comment:
Namespaces and service name should be pulled from variables as they can change depending on the environment.